### PR TITLE
feat(Icon): add `lookSmaller` boolean props

### PR DESCRIPTION
**Pull request by patrickliu-sa**
Add `lookSmaller` boolean prop to show reduced icon in display within original icon size.

icon-looksmaller

EXAMPLE USAGE:

```js
import {
  Icon
} from 'seek-asia-style-guide/react';

<Icon
  lookSmaller
  size="normal"
  type="birthdate"
/>
```

### CODE REVIEW

**Comment by patrickliu-sa** [comment]
> 🤔 Is it _always_ better to have `displaySize` one smaller than `size`? Is it possible to just enforce that and not need the `displaySize` prop?

It can be an icon size of large with displaySize of small 😄 , but that might be a very edge case. And Joanne also mentioned that it won't be gonna happen.

Or we can try to use `bool` on `smallerDisplaySize` prop, perhaps?

**Comment by mengtzu** [comment]
I am coming around to the opinion that it's better to eliminate edge cases than support them 😂 

The boolean for smaller display sounds like it would be easier to understand?  Maybe straw poll with a few random devs to see what makes more sense.

**Comment by patrickliu-sa** [comment]
After discussed with @CedricPoon-SeekAsia we decided to drop `displaySize` and rename `smallerDisplaySize` to `lookSmaller` for easy understanding

File: react/Icon/Icon.less [hunk 1]
```diff
@@ -11,21 +11,38 @@
...
+  .sqSize(@grid-base * 4);
 }
 
 .sizeNormal {
-  width: @grid-base * 6;
-  height: @grid-base * 6;
+  .sqSize(@grid-base * 6);
+  &.lookSmaller {
+    svg {
+      .sqSize(@grid-base * 4) !important;
```
Comment (ChanJit) [inline react/Icon/Icon.less]: 😂 really need to use `!important`?

File: react/Icon/Icon.js [hunk 2]
```diff
@@ -25,6 +26,7 @@ export default function Icon({
         [styles.sizeSmall]: size === 'small',
         [styles.sizeNormal]: size === 'normal',
         [styles.sizeLarge]: size === 'large',
+        [styles.lookSmaller]: lookSmaller,
```
Comment (ChanJit) [inline react/Icon/Icon.js]: why dont add a size name `xs` or `extraSmall`?

Comment (patrickliu-sa) [reply react/Icon/Icon.js]: Attempted to have one scale down only. Avoid misuse of the case like "Large size icon with small display size" 😸

File: react/Icon/Icon.less (hunk 1 above)
Comment (patrickliu-sa) [reply react/Icon/Icon.less]: Because svg path need to set their width and height as `100%` for initial value, or it will shrink as `0%` by default.

Comment (mengtzu) [reply react/Icon/Icon.less]: Probably best to comment it as such then :)

File: react/Icon/Icon.js (hunk 2 above)
Comment (ChanJit) [reply react/Icon/Icon.js]: the svg size is 16 px X 16 px right? so it should be same as small, why dont encourage them use small?

Comment (patrickliu-sa) [reply react/Icon/Icon.js]: Then the spacing around icon should be set within the icon consumer, not from the icon component itself...

**Comment by mengtzu** [comment]
Implementation looks fine to me, I'm not sure about the name, but if that's the one everyone agrees on... 😂

File: react/Icon/Icon.less [hunk 3]
```diff
@@ -1,31 +1,51 @@
 @import (reference) '~seek-asia-style-guide/theme';
 
+/* [TO DO] use maps when LESS v3.5 available */
```
Comment (kosanna) [inline react/Icon/Icon.less]: lol, yeah, that's why I like sass :)

File: react/Icon/Icon.sketch.less [hunk 4]
```diff
@@ -16,3 +16,7 @@
   width: 22px;
   height: 22px;
 }
+
+.whiteCanvas {
+  background-color: #ffffff;
```
Comment (kosanna) [inline react/Icon/Icon.sketch.less]: Do we have a variable for styleguide white?

**Comment by kosanna** [comment]
What is the idea of it if we are just downgrading one size? Can't we just use respective size directly?

File: react/Icon/Icon.sketch.less (hunk 4 above)
Comment (patrickliu-sa) [reply react/Icon/Icon.sketch.less]: ah yes, `@saWhite` 🤦‍♂️

**Comment by kosanna** [comment]
Agreed to close for now as potentially might be refactored on Monday

**Pull request closed without merging**